File: Java/src/main/java/com/gk/java/PalindromeNumber.java
package com.gk.java;

public class PalindromeNumber {
    public static void main(String[] args) {
//            System.out.println("100 is palindrome ="+isPalindrome(100));
        System.out.println("101 is palindrome =" + isPalindrome(101));
        System.out.println("101 is palindrome =" + checkPalindromeWithString(101));
        System.out.println("165 is palindrome =" + checkPalindromeWithString(165));
        System.out.println("161 is palindrome =" + checkPalindromeWithInbuilt(161));
        System.out.println("169 is palindrome =" + checkPalindromeWithInbuilt(169));
    }

    static boolean isPalindrome(int num) {
        int sum = 0, r, temp;
        temp = num;
        while (num > 0) {
            r = num % 10;
            System.out.println("r =" + r);
            sum = (sum * 10) + r;
            System.out.println("sum =" + sum);
            num = num / 10;
            System.out.println("num = " + num);
        }
        if (temp == sum) {
            return true;
        }

        return false;
    }

    // We can use same logic for String reverse
    static boolean checkPalindromeWithString(int num) {
        int temp = num;
        String rev = "";
        String strNum = Integer.toString(num);
        for (int i = strNum.length(); i > 0; --i) {
            rev = rev + strNum.charAt(i-1);
        }
        return temp == Integer.parseInt(rev);
    }

    // We can use same logic for String reverse
    static boolean checkPalindromeWithInbuilt(int num){
        StringBuilder builder =new StringBuilder(String.valueOf(num));
        return num == Integer.parseInt(builder.reverse().toString());
    }
}

File: Koltin/src/main/java/com/gk/koltin/PalindromeNumber.kt
package com.gk.koltin

fun main() {
    println("${656.isPalindrome()}")
    println("${650.isPalindrome()}")
    println("${"656".isPalindrome()}")
    println("${"656".isPalindromeReversed()}")
    println("${"650".isPalindromeReversed()}")


    5 addition 5
}

infix fun Int.addition(b:Int){
    println("${this+b}")
}

private fun Int.isPalindrome(): Boolean {
    var rev: Int
    var sum: Int = 0
    var temp = this
    while (temp > 0) {
        rev = temp % 10
        sum = (sum * 10) + rev
        temp /= 10
    }
    return sum == this
}

// We can use same logic for String reverse
private fun String.isPalindrome(): Boolean {
    var rev = ""
    for (i in this.length downTo 1) {
        rev += this[i - 1]
    }
    return this == rev
}

private fun String.isPalindromeReversed():Boolean{
    val temp=this
    return temp == reversed()
}